### ISSUE TITLE
Use sass modules

### ISSUE THREAD

**Issue by li-kai** [nusmodifications/nusmods#2213]
### Is your feature request related to a problem? Please describe.

SASS has introduced a new feature called `@use` that does what we want instead of `@import` which is prone to duplicating styles.

https://css-tricks.com/introducing-sass-modules/

### Describe the solution you'd like

Replace instances of using `@import` in most places with `@use` where possible.

e.g. `@import '~styles/utils/modules-entry';` should be `@use '~styles/utils/modules-entry';`

### Describe alternatives you've considered

NIL. I've not explored sass modules too much, but it looks powerful for other things than just importing.

### Additional context

https://sass-lang.com/documentation/cli/migrator
http://sass.logdown.com/

**Comment by fernandacanepa** [comment]
Hi! I read the link you left and indeed it looks powerful. All that you want here is to replace the @imports with @use? think I can do that haha

**Comment by li-kai** [comment]
Yes, that's all we want at the moment. I'll assign the issue if you want to take this up!

**Comment by peironggg** [comment]
Hello! Is this issue taken? If not, I would like to have a go at it !

**Comment by fernandacanepa** [comment]
> Yes, that's all we want at the moment. I'll assign the issue if you want to take this up!

Yass! I would like to take it!

**Comment by li-kai** [comment]
All yours! Thanks for volunteering for this.

**Comment by ravern** [comment]
Hey, we're doing some clean ups over the summer. Closing this for now because this issue is stale. Feel free to re-open this if you think otherwise!
